File: Java/Platform/Communication/HTTPClientTest/ItemInfo.java
record ItemInfo(double price, int stock) {

	public static ItemInfo parse(String text) {
		String[] parts = text.split(",");
		double p = Double.parseDouble(parts[0].substring(6));
		int s = Integer.parseInt(parts[1].substring(6));
		return new ItemInfo(p, s);
	}

}

File: dotnet/Platform/Communication/HttpClientTest/DemoApp/ItemInfo.cs
readonly record struct ItemInfo(double Price, int Stock)
{
    public static ItemInfo Parse(string source)
    {
        string[] parts = source.Split(',');
        double p = double.Parse(parts[0].Substring(6));
        int s = int.Parse(parts[1].Substring(6));
        return new ItemInfo(p, s);
    }
}
